### PR TITLE
T1492

### PR DESCRIPTION
**Pull request by secretrobotron**

### CODE REVIEW

File: cornfield/config/default.json [hunk 1]
```diff
@@ -14,5 +14,17 @@
...
+    "test": "../templates/test.html",
+    "robots": "../templates/supported/robots/index.html",
+    "report": "../templates/supported/report/index.html",
+    "popup": "../templates/supported/popup/index.html",
+    "musicvideo": "../templates/supported/musicvideo/index.html",
+    "context": "../templates/supported/context/index.html",
+    "robots": "../templates/supported/newscaster/index.html"
+  },
+  "exportAssets": [
+    "../../dist/buttered-popcorn.min.js"
```
Comment (humphd) [inline cornfield/config/default.json]: In #1486, with the storycamp target, we're changing this slightly:

node make storycamp

This makes buttered-popcorn.min.js, renames it popcorn.js and puts it in dist/external/popcorn-js/popcorn.js

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/e6c4f2b0c094965/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/e6c4f2b0c094965/output.txt

Total script time: 0.29 mins

File: cornfield/app.js [hunk 2]
```diff
@@ -84,20 +87,81 @@ function publishRoute( req, res ){
...
+              bodyEndTagIndex,
+              externalAssetsString = '',
+              popcornString = '',
+              currentMedia,
+              currentTrack,
+              currentTrackEvent,
+              mediaPopcornOptions,
+              j, k;
+
+          data = data.replace( /\s*<script[\.\/='":_-\w\s]*data-butter-exclude[\.\/='":_-\w\s]*><\/script>/g, '' );
```
Comment (humphd) [inline cornfield/app.js]: Throw a comment above this line so it's more clear what the regex wants to do.

File: cornfield/config/default.json [hunk 3]
```diff
@@ -15,5 +15,16 @@
     "templates": "../templates",
     "publish": "view",
     "publishPrefix": "http://localhost:8888/cornfield/view"
-  }
+  },
+  "templates": {
+    "test": "../templates/test.html",
```
Comment (humphd) [inline cornfield/config/default.json:20]: Get rid of musicvideo and context from this list.

File: cornfield/app.js [hunk 4]
```diff
@@ -10,7 +10,9 @@ const express = require('express'),
       TEMPLATES_DIR =  CONFIG.dirs.templates,
       PUBLISH_DIR = CONFIG.dirs.publish,
       PUBLISH_PREFIX = CONFIG.dirs.publishPrefix,
-      WWW_ROOT = path.resolve(CONFIG.dirs.wwwRoot || __dirname + "/..");
+      WWW_ROOT = path.resolve(CONFIG.dirs.wwwRoot || __dirname + "/.."),
```
Comment (humphd) [inline cornfield/app.js]: We should get in the habit of using path.join in node.js code that builds paths.

File: external/popcorn-js [hunk 5]
```diff
@@ -1 +1 @@
-Subproject commit da087cb2107af5ac0b22180cc80f9033cb7a8647
+Subproject commit 4f8dfcfc3036f27cabb2b20024e51f2167dca127
```
Comment (humphd) [inline external/popcorn-js]: I assume this is intentional, but want to confirm.

**Pull request merged**